### PR TITLE
Add reducers

### PR DESCRIPTION
**Pull request by Ngufor-emmanuella**
in this project I did the following
-Add Redux Toolkit
-Define action types for adding and removing a book.
-Set the initial state to be an empty array of books.
-Export Action Creators for your actions.
-Write reducer and export it as default.
-Define state changes for the actions created.
- return the current state.

### CODE REVIEW

File: src/redux/Books/books.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+});
+
+// create reducers
+const initialState = [];
+
+const bookReducer = (state = initialState, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case ADD_BOOK:
+      return state.filter((books) => books.id !== payload);
```
Comment (levy002) [inline src/redux/Books/books.js]: - [ ] Kindly make sure to add a new object in this case of `ADD_BOOK`. We only filter only when we are removing the book but by adding a new book please make sure you are adding a new object of the newly added book.

File: src/redux/Books/books.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    default:
+      return state;
+  }
+};
```
Comment (levy002) [inline src/redux/Books/books.js]: - [ ] Kindly make sure the case for `REMOVE_BOOK` is presented in your reducer. This book reducer is in charge of adding and removing books in your store.

**Review by levy002 — CHANGES_REQUESTED** [review]
## Hello @Ngufor-emmanuella   , :wave:
### I'm Levy, your PR reviewer

Good job so far :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!



## Status: REQUIRED CHANGES ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @levy002 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/Books/books.js (hunk 1 above)
Comment (Ngufor-emmanuella) [reply src/redux/Books/books.js]: thank you please a little clarification here am I supposed to use the map mehod in place of filter?

Comment (levy002) [reply src/redux/Books/books.js]: Hello @Ngufor-emmanuella, When adding a new book we don't really need to iterate through our array, we only need to insert a new book object into our array containing all books. That action will not require iterating through the array.

Comment (Ngufor-emmanuella) [reply src/redux/Books/books.js]: ok thanks

**Review by ShadyShawkat — APPROVED** [review]
## Hi @Ngufor-emmanuella, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**